### PR TITLE
Refactor the resolution of imports to be spec compliant 

### LINKED ISSUE #242: Imports are not spec compliant
The [official spec about imports](https://sass-lang.com/documentation/at-rules/import#finding-the-file) describes this algorithm:

1. try to resolve the file relatively to the current file
2. if failed, try to resolve the file using load paths (what we call "import paths" in scssphp)

scssphp does not implement that at all. It never resolves files relatively. What it does instead is prepending some load paths in the configured list when entering an imported file and removing it when leaving. And by default, it puts the empty string as first load path in the list (which leads to resolving to `getcwd()` due to the way PHP resolves relative paths). But this one can be removed (and currently, `pscss -I /my/load/path` removes it)
This has several drawbacks:
- relative imports for the root stylesheet work only if you take care of doing `chdir(dirname($path));` before the compilation or if you add it as a load path. this is quite inconvenient
- once you start importing files, you leak load paths for the imports done by these files. this allows a file to resolve imports relatively to its importing file (and the importing file of the importing file, etc...)
- if you catch a compilation exception and then reuse the Compiler for another compilation, these leaked load paths are still there and are the first ones...
- by doing `$compiler->setImportPaths([]);`, you can effectively disable `@import` as there won't be any resolution done (and this is even [documented](https://github.com/scssphp/scssphp/blob/51e7484d2b590031ffa5fbabddea7682e534c88b/docs/docs/index.md#import-paths)

see https://github.com/sass/sass-spec/pull/1589 for a spec failing on this.

### PR DESCRIPTION
**Pull request by stof**
Closes #242

### CODE REVIEW

File: src/Compiler.php [hunk 1]
```diff
@@ -5336,13 +5336,124 @@ public function findImport($url)
...
         }
 
-        if ($urls) {
-            if (! $hasExtension || preg_match('/[.]scss$/', $url)) {
-                throw $this->error("`$url` file not found for @import");
-            }
+        throw $this->error("`$url` file not found for @import");
+    }
+
+    private function resolveImportPath($url, $baseDir)
```
Comment (stof) [inline src/Compiler.php:5342]: This method (and the following helper methods) are implementing the logic at https://github.com/sass/dart-sass/blob/28a582f45e3f9f3384071a83db2012b4e2e16707/lib/src/importer/utils.dart#L41-L44 with 2 exceptions (not counting the fact that dart-sass makes the `path.join()` in the caller):

- support for the `.sass` extension, as we don't support parsing them (this means we don't consider them as a source of conflict either)
- support for `.import.scss` files, as that's part of implementing Sass modules (this one will become support when/if we implement modules, but the code will probably have changed quite a bit at that point anyway)

**Comment by stof** [comment]
Regression tests for our broken implementation of relative imports have been accepted in sass-spec itself: https://github.com/sass/sass-spec/pull/1589 :tada:

File: src/Compiler.php [hunk 2]
```diff
@@ -5336,13 +5336,124 @@ public function findImport($url)
...
+    }
+
+    /**
+     * @param string $path
+     *
+     * @return string[]
+     */
+    private function tryImportPath($path)
+    {
+        $partial = dirname($path).'/_'.basename($path);
```
Comment (Cerdic) [inline src/Compiler.php:5408]: Is it the expected behaviour that `@import '_partial';`  will try to import  `__partial.scss`?

I had the feeling that an explicit partial reference in the `@import` should just be taken as is, not trying to partialize the already partial file (at least that's what scssphp was doing before).
Am I wrong?

Comment (stof) [reply src/Compiler.php:5408]: dart-sass will do it too.

Comment (Cerdic) [reply src/Compiler.php:5408]: yep seems more a flaw in the design than a real feature :p 
but if dart-saas do that, let's stick on it

This seems all good for me besides that.

Comment (stof) [reply src/Compiler.php:5408]: I opened https://github.com/sass/sass/issues/2940 to clarify that.

Comment (stof) [reply src/Compiler.php:5408]: @Cerdic it's `dart-sass`, not `dart-saas` :smile:

Comment (stof) [reply src/Compiler.php:5408]: The Sass team answered that this behavior is the same in dart-sass and ruby-sass, and not worth a BC break to undo it, even though they would have preferred doing that differently retrospectively (that was probably not intended). So I think it is good we follow the same rule.

**Pull request merged**